### DISCUSSION TITLE
Get next element

### DISCUSSION THREAD

**Discussion by JohnnyQQQQ** [MontFerret/ferret#668] [Q&A] [answered]
I try to scrape the "Age" of the following DOM
```html
<div id="container">
  <span>Name</span>
  <span>Peter Parker</span>
  <span>Age</span>
  <span>23</span>
</div>
```

The problem is, I don't have any ID or class I can reference. My question is, how is it possible to access the age? The amount of spans is always varrying.

My idea was something like this:

```
LET adminstrationAgeRaw = (
    FOR el IN ELEMENT(doc, '#container')
        FILTER el.text() == 'Age'
        RETURN el.next().text()
)
LET adminstrationAge = FIRST(adminstrationAgeRaw )
```

I couldn't find any direction to a solution regarding the syntax by looking into the docs, as there is not much regarding this topic.

**Comment by ziflex** [comment, marked as answer]
Try something like this

```
LET elements = ELEMENTS(doc, '#container span')

FOR el, i IN elements
    FILTER ROUND(i/2) == i/2
    
    LET key = el.innerText
    LET sibling = elements[i+1]
    LET value = sibling.innerText
    
    RETURN { [text]: value }
```

**Comment by JohnnyQQQQ** [comment]
@ziflex thanks alot for the quick answer, works perfectly fine!
